functions.sh

```diff
@@ -49,3 +49,15 @@ docker_try_rmi() {
         docker rmi $image_name
     }
 }
+
+sonic_get_version() {
+    local describe=$(git describe --tags)
+    local latest_tag=$(git describe --tags --abbrev=0)
+    BUILD_NUMBER=${BUILD_NUMBER:-dev}
+    ## Cehck if we are on tagged commit
+    if [ "$describe" == "$latest_tag" ]; then
+        echo "$latest_tag"
+    else
+        echo "${latest_tag}.${BUILD_NUMBER}-$(git rev-parse --short HEAD)"
```

Comment:
in case we are not on tagged commit, I am not sure how useful the latest_tag is useful here, maybe it is better to use the branch name? 

```
git rev-parse --abbrev-ref HEAD
```